tests/core/test_periods.py

```diff
@@ -1,114 +1,151 @@
 # -*- coding: utf-8 -*-
 
 from __future__ import unicode_literals, print_function, division, absolute_import
-from nose.tools import assert_equal, raises
 
-from openfisca_core.periods import Period, Instant, YEAR, MONTH, period
+import pytest
+
+from openfisca_core.periods import Period, Instant, YEAR, MONTH, DAY, period
 from openfisca_core.commons import to_unicode
 
 first_jan = Instant((2014, 1, 1))
 first_march = Instant((2014, 3, 1))
 
 
-# Test Period -> String
+'''
+Test Period -> String
+'''
+
+
+# Years
 
 def test_year():
-    assert_equal(to_unicode(Period((YEAR, first_jan, 1))), '2014')
+    assert to_unicode(Period((YEAR, first_jan, 1))) == '2014'
 
 
 def test_12_months_is_a_year():
-    assert_equal(to_unicode(Period((MONTH, first_jan, 12))), '2014')
+    assert to_unicode(Period((MONTH, first_jan, 12))) == '2014'
 
 
 def test_rolling_year():
-    assert_equal(to_unicode(Period((MONTH, first_march, 12))), 'year:2014-03')
-    assert_equal(to_unicode(Period((YEAR, first_march, 1))), 'year:2014-03')
+    assert to_unicode(Period((MONTH, first_march, 12))) == 'year:2014-03'
+    assert to_unicode(Period((YEAR, first_march, 1))) == 'year:2014-03'
+
+
+def test_several_years():
+    assert to_unicode(Period((YEAR, first_jan, 3))) == 'year:2014:3'
+    assert to_unicode(Period((YEAR, first_march, 3))) == 'year:2014-03:3'
+
 
+# Months
 
 def test_month():
-    assert_equal(to_unicode(Period((MONTH, first_jan, 1))), '2014-01')
+    assert to_unicode(Period((MONTH, first_jan, 1))) == '2014-01'
 
 
 def test_several_months():
-    assert_equal(to_unicode(Period((MONTH, first_jan, 3))), 'month:2014-01:3')
-    assert_equal(to_unicode(Period((MONTH, first_march, 3))), 'month:2014-03:3')
+    assert to_unicode(Period((MONTH, first_jan, 3))) == 'month:2014-01:3'
```

Comment:
what's the reasoning for moving to `assert` instead of `assert_equal`? 

I liked the output from `asset_equal` when it fails better than the built-in python `assert` failed. `assert_equal` tells you what the two values where, but `assert` only says it failed.

Reply:
Hi! Only that wére moving from nose to pytest, so I just did a bit of cleanup.

Reply:
There's this PR #746 that introduces the change, notably because `nose` it in maintenance mode.